model/item.cc

```diff
@@ -223,8 +223,12 @@ namespace minsky
   bool Item::visibleWithinGroup() const 
   {return !attachedToDefiningVar();}
 
+  bool Item::inputDefined() {
+    if(m_ports.size()==0) return false;
```

Comment:
.empty() is considered better, although for vectors, which this is, it is basically equivalent anyway.